source/common/secret/secret_manager_impl.cc

```diff
@@ -37,5 +38,37 @@ TlsCertificateConfigProviderSharedPtr SecretManagerImpl::createInlineTlsCertific
   return std::make_shared<TlsCertificateConfigProviderImpl>(tls_certificate);
 }
 
+void SecretManagerImpl::removeDynamicSecretProvider(const std::string& map_key) {
+  ENVOY_LOG(debug, "Unregister secret provider. hash key: {}", map_key);
+
+  RELEASE_ASSERT(dynamic_secret_providers_.erase(map_key) == 1, "");
```

Comment:
Thanks for pointing this out. Moved erase() call outside of RELEASE_ASSERT.

RELEASE_ASSERT was recommended here [comment](https://github.com/envoyproxy/envoy/pull/4176#discussion_r211782555)

Reply:
> This should be RELEASE_ASSERT; otherwise in opt builds, this entire line disappears.

With it split into two lines, the rational for making it RELEASE_ASSERT instead of ASSERT is no longer true, so this should now be ASSERT